agent/config/config.go

```diff
@@ -311,7 +311,16 @@ func environmentConfig() (Config, error) {
 	taskENIEnabled := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_ENI"), false)
 	taskIAMRoleEnabled := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_IAM_ROLE"), false)
 	taskIAMRoleEnabledForNetworkHost := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST"), false)
-	taskCPUMemLimitEnabled := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_CPU_MEM_LIMIT"), true)
+
+	var taskCPUMemLimitEnabled Conditional
+	taskCPUMemLimitConfigString := os.Getenv("ECS_ENABLE_TASK_CPU_MEM_LIMIT")
+
+	switch {
+	case utils.ParseBool(taskCPUMemLimitConfigString, true) && utils.ParseBool(taskCPUMemLimitConfigString, false):
```

Comment:
This is going to be confusing for folks without the context. 
A comment to explain the logic would be helpful imho. 

Also, isn't this check a bit excessive.

Reply:
Yes, I over-thought this.  I'll make it more clear